### PR TITLE
Fixed Issue #20 [Feature Request] Add loss functions for classification tasks

### PR DESCRIPTION
**Pull request by arijitde92**
Closes: #20

- **Title** : Add loss functions for classification tasks
- **Name:** Arijit De
- **Idenitfy yourself:** SSoC'24 Contributor

### Describe the add-ons or changes you've made 📃

Added implementations of hinge loss, KL Divergence loss and four versions of cross entropy loss from scratch using numpy.
The implementations also contains results with some sample data.
Also updated the README.md file inside ML directory with the links to the directories of the said loss functions.

### Type of change ☑️

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested? ⚙️

It has been tested with sample data and the test code is also available in the respective loss function files.

### Checklist: ☑️

- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/AI-Code/blob/main/README.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/AI-Code/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.

### Screenshots 📷

Cross Entropy loss
CE_Loss

Hinge Loss
hinge_loss

KL-Divergence Loss
Screenshot 2024-06-11 185102

### CODE REVIEW

File: ML/Algorithms/Losses/HingeLoss/hinge_loss.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    if (y_true is not None) and (y_pred is not None):
+        if type(y_true) == list:
+            y_true = np.asarray(y_true)
+        if type(y_pred) == list:
+            y_pred = np.asarray(y_pred)
+        assert y_true.shape[0] == y_pred.shape[0], f"Batch size of y_true ({y_true.shape[0]}) does not match y_pred ({y_pred.shape[0]})"
+    
+    # replacing 0 values to -1
+    y_pred = np.where(y_pred == 0, -1, 1)
+    y_true = np.where(y_true == 0, -1, 1)
```
Comment (Avdhesh-Varshney) [inline ML/Algorithms/Losses/HingeLoss/hinge_loss.py:21]: why we have replacing some values from 0 to -1? explain it here.

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
@arijitde92 LGTM.

File: ML/Algorithms/Losses/HingeLoss/hinge_loss.py (hunk 1 above)
Comment (arijitde92) [reply ML/Algorithms/Losses/HingeLoss/hinge_loss.py:21]: Hinge loss acts as a maximum margin classifier (like SVM) and according to [Wikipedia](https://en.wikipedia.org/wiki/Hinge_loss) of Hinge Loss, the mathematical definition states that the label values should be 1 and -1.
See below screenshot
image

**Review by Avdhesh-Varshney — APPROVED** [review]
@arijitde92 PR Approved 🎉
